core/storage/blog/gae_models.py

```diff
@@ -0,0 +1,424 @@
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostSummaryModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.author_id == user_id
+        ).get(keys_only=True) is not None
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model data has already been associated as a part of the
+        BlogPostModel to the user and thus does not need a separate user
+        association.
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data corresponding to a user (author_id), but this
+        isn't exported because noteworthy details that belong to this
+        model have already been exported as a part of the BlogPostModel.
+        """
+        return dict(super(BlogPostSummaryModel, cls).get_export_policy(), **{
+            'author_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'title': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'summary': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'url_fragment': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'tags': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'thumbnail_filename': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'published_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @classmethod
+    def get_blog_post_summary_models(cls, blog_post_ids):
+        """Returns a list of BlogPostSummaryModels for blog posts created by
+        the user.
+
+        Args:
+            blog_post_ids: list(str). Contains blog post ids for which the blog
+                post summary models are to be fetched.
+
+        Returns:
+            list(BlogPostSummaryModel). The list of BlogPostSummaryModels
+            fetched using blog_post_ids.
+        """
+        return cls.get_multi(blog_post_ids)
+
+
+class BlogPostRightsModel(base_models.BaseModel):
+    """Storage model for rights related to a blog post.
+
+    The id of each instance is the blog_post_id.
+    """
+
+    # The user_ids of the blog editors of this blog post.
+    editor_ids = datastore_services.StringProperty(indexed=True, repeated=True)
+
+    # Whether this blog post is published or not.
+    # False if blog post is a draft, True if published.
+    blog_post_is_published = datastore_services.BooleanProperty(
+        indexed=True, required=True, default=False)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize or delete corresponding
+        to a user: editor_ids field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
```

Comment:
Why `LOCALLY_PSEUDONYMIZE`? In the wipeout service, you're removing the user ID from `editor_ids` but not pseudonymizing it. Should this be `DELETE`?

Reply:
done.